core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -732,6 +732,14 @@ class KafkaApis(val requestChannel: RequestChannel,
         response.responseData.asScala.foreach { case (topicPartition, data) =>
           // record the bytes out metrics only when the response is being sent
           brokerTopicStats.updateBytesOut(topicPartition.topic, fetchRequest.isFromFollower, data.records.sizeInBytes)
+          // record any outgoing reassignment rate if the fetch request came from an addingReplica of the partition
+          if (fetchRequest.isFromFollower) {
+            replicaManager.getPartition(topicPartition) match {
+              case HostedPartition.Online(p) => if (p.isAddingReplica(fetchRequest.replicaId))
```

Comment:
I guess we didn't discuss this but it raised a point in my mind - do we want to count an adding replica that is in ISR as reassignment traffic? I believe yes, since we wouldn't have that traffic were we not reassigning

Reply:
Yea I also think it might make more sense to count it as reassignment traffic. Also it might fall back with replication etc. which would make our life easier.